### DISCUSSION TITLE
Instable, stop responding after 3-10 days

### DISCUSSION THREAD

**Discussion by tomasenskede** [jomjol/AI-on-the-edge-device#2880] [General]
### The Problem

I've used AI on the edge for my watermeter since January 17. I've encountered stops on January 19, January 23, January 27, and February 8. Is this frequency of stops normal, and is there a way to enhance its stability? I recently updated to version 15.5.0 today, anticipating improved stability. Could there be another factor causing these stops?

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/0993b8bf-8821-4a66-9743-2e718da45222)
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/89a205a4-75aa-4076-aeb9-34f0c4e18b5d)

### Version

15.5

### Logfile

[0d00h00m25s] 2024-02-08T20:53:32 <INF> [MAIN] Device info: CPU cores: 2, Chip revision: 101
[0d00h00m25s] 2024-02-08T20:53:32 <INF> [MAIN] SD card info: Name: SL16G, Capacity: 15193MB, Free: 15152MB
[0d00h00m27s] 2024-02-08T20:53:34 <INF> [MAIN] Initialization completed successfully
[0d00h00m28s] 2024-02-08T20:53:34 <INF> [WIFI] Connected to: HermanssonHD, RSSI: -89
[0d00h00m30s] 2024-02-08T20:53:36 <INF> [LOGFILE] Set log level to ERROR
[0d00h01m25s] 2024-02-08T20:54:32 <ERR> [POSTPROC] main: Raw: 01503.6445, Value: , Status: Neg. Rate - Read: - Raw: 01503.6445 - Pre: 1504.6445
[0d00h06m25s] 2024-02-08T20:59:30 <ERR> [POSTPROC] main: Raw: 01503.6445, Value: , Status: Neg. Rate - Read: - Raw: 01503.6445 - Pre: 1504.6445
[0d00h11m25s] 2024-02-08T21:04:30 <ERR> [POSTPROC] main: Raw: 01503.6463, Value: , Status: Neg. Rate - Read: - Raw: 01503.6463 - Pre: 1504.6445
[0d00h16m25s] 2024-02-08T21:09:30 <ERR> [POSTPROC] main: Raw: 01503.6475, Value: , Status: Neg. Rate - Read: - Raw: 01503.6475 - Pre: 1504.6445
[0d00h21m25s] 2024-02-08T21:14:30 <ERR> [POSTPROC] main: Raw: 01503.6568, Value: , Status: Neg. Rate - Read: - Raw: 01503.6568 - Pre: 1504.6445

 
Reload Show Full Log Show Older Log Files
```


### Expected Behavior

_No response_

### Screenshots

_No response_

### Additional Context

_No response_

**Comment by SybexX** [comment]
As I understood you, you used v15.4.0, maybe it could be related to this https://github.com/jomjol/AI-on-the-edge-device/issues/2823 or your very poor WiFi reception "Connected to: HermanssonHD, RSSI: -89"

**Comment by caco3** [comment]
Please provide a full debug log containing what ever you mean with "stopping".
And please update to 15.6.0 as we fixed a critical bug.

**Comment by tomasenskede** [comment]
New issue;

I did update to 15.6.0 and now I have 7505.0153 (!) why?
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/be48f1a4-3229-49c3-ae33-f766b755e78d)

**Comment by SybexX** [comment]
That's the PreValue, apparently he recognized the 1 as 7 during the previous reading.
The currently read value is correct.

**Comment by friedpa** [comment]
Just set the PreValue again and use MaxRate for preventing such huge steps in the reading

**Comment by SybexX** [comment]
Doesn't the system do it by itself because PreValue is too old?

**Comment by caco3** [comment]
No, the PreValue is just used for a plausibility check. See https://jomjol.github.io/AI-on-the-edge-device-docs/Correction%20Algorithm/#prevalue

**Reply by SybexX** [reply]
I thought that the setting "PreValueAgeStartup" resets the PreValue

**Reply by caco3** [reply]
hmm, good point. I actually do not know what happens after `PreValueAgeStartup` expires. i also would expect that the `PreValue` in such case gets ignored, but will it then simply use the new raw value' I don't know.

**Reply by SybexX** [reply]
habe mal nachgeguckt:
wenn ich nichts übersehen habe, sollte PreValue bei der ersten Runde mit den gelesenen Wert ersetzt werden, wenn AllowNegativeRates aktiviert ist. Ist AllowNegativeRates deaktiviert, muß PreValue manuel gesetzt werden, wegen if (!NUMBERS[j]->AllowNegativeRates){}

**Comment by tomasenskede** [comment]
Lets see if it reads correct from now...
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/aee62a0d-ef7f-4153-8d25-0e20fb352546)

**Comment by tomasenskede** [comment]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/76ae3d16-1010-4bab-ab46-d6046840dc8a)

**Reply by SybexX** [reply]
Did you set the RSSI Threshold to 0 because of your poor WiFi?

**Reply by tomasenskede** [reply]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/ce74a738-269f-4e25-a77e-1b43c0f4b688)

yes!

**Comment by tomasenskede** [comment]
Neg. Rate - Read: - Raw: **01506.8384** - Pre: **55070.6175**
55050.6175 previous value

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/20cf0dc9-a2db-429c-a0ef-1a34998e6256)

**Reply by Slider0007** [reply]
@tomasenskede:
All your ROIs are not perfectly placed. Please readjust your ROIs according the documentation. There is a quite detailed section for that. 

Also it's benefical for stability to set alignment around the center of image, not just on one side. Alternative: Switch off Alignment Algo (expert Parameter) to 'off', then only initial rotation anlge alignment (defined in reference image page) will be applied.

**Reply by tomasenskede** [reply]
Isnt this correct?
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/f28c789a-22e1-4bc9-84f1-0b140214654d)

**Reply by tomasenskede** [reply]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/b15bea92-2950-43a7-b58d-4fc14ea9a8ea)
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/93a16c0c-ec71-43dc-8595-cb53155ac54e)
Like this?

**Reply by SybexX** [reply]
try reducing the brightness

**Reply by SybexX** [reply]
and since the distance between your Digit-Rois is the same, if I were you I would activate the "Keep equidistance of" option

**Reply by tomasenskede** [reply]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/99ade424-d0ed-47a9-9849-948dc34aeffd)

**Reply by SybexX** [reply]
try Brightness -2, Contrast 2 and Saturation -2

**Reply by tomasenskede** [reply]
"Keep equidistance of" ? where do I find this option?

**Reply by SybexX** [reply]
![303932195-f28c789a-22e1-4bc9-84f1-0b140214654d](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/35d34193-e6f6-4a5d-a6ef-95358f999ec8)

**Reply by caco3** [reply]
Your ROIs are too small, check the documentation for examples!

**Reply by tomasenskede** [reply]
Neg. Rate - Read: - Raw: **01507.1673** - Pre: **54864.6060**

Again

**Reply by caco3** [reply]
> Neg. Rate - Read: - Raw: 01507.1673 - Pre: 54864.6060

Well, we told you already that you have to set the initial PreValue.
But first you have to set the Reference Image, marks and ROIs properly!

**Comment by tomasenskede** [comment]
Now it has been running stable until yesterday evening
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/40f9ed55-2278-4e15-952c-983f6d97a8c6)

Current readout... 
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/150d4bae-4024-456a-85aa-7176f2662976)

The aligment isnt as good as in the 
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/e5163829-a5ac-4638-bd76-002aa064d5c6)
and
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/32595351/b2d42699-5ce2-4360-b9b8-3eb5aede3fc3)

**Reply by caco3** [reply]
The reference marks should be as far apart as possible. Also the one around the `m3` should be smaller.
I suggest to use the `2.5` on the top and add a sticker at the bottom.

The early transition are caused by a bug in 15.5 resp. 15.6 and will be reverted in the next release

**Comment by caco3** [comment]
We just released [v15.7.0](https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v15.7.0) which hopefully fixes this issue.
